frappe/desk/doctype/tag/tag.py

```diff
@@ -81,7 +81,7 @@ def update(self, dn, tl):
 		if not tl:
 			tags = ''
 		else:
-			tl = list(set(filter(lambda x: x, tl)))
+			tl = list(dict.fromkeys(filter(lambda x: x, tl)))
```

Comment:
You can use `unique` in `data.py` https://github.com/frappe/frappe/blob/f2744a993b5c7466d1574d2d8924468ba4c9db95/frappe/utils/data.py#L1043

Reply:
apparently `dict.fromkeys()` is used to preserve the order, but that only works on Python >= 3.7

Reply:
unique works, yeah

Reply:
Yeah.. it works